go/enclave/components/rollup_consumer.go

```diff
@@ -90,6 +90,21 @@ func (rc *rollupConsumerImpl) ProcessRollup(ctx context.Context, rollup *common.
 		rc.logger.Warn("Skipping rollup because it was compressed on top of a non-canonical block", "block_hash", rollup.Header.CompressionL1Head, log.RollupHashKey, rollup.Hash(), log.ErrKey, err)
 		return nil, nil
 	}
+
+	// if all batches included in this rollup exist, we don't need to process
```

Comment:
this comment suggests you wanted to return early if the last hash matches but doesn't look like the early return is there?

Reply:
oops. good shout